autogen/oai/client.py

```diff
@@ -148,10 +353,28 @@ def _client(self, config: Dict[str, Any], openai_config: Dict[str, Any]) -> Open
             if openai_config["azure_deployment"] is not None:
                 openai_config["azure_deployment"] = openai_config["azure_deployment"].replace(".", "")
             openai_config["azure_endpoint"] = openai_config.get("azure_endpoint", openai_config.pop("base_url", None))
-            client = AzureOpenAI(**openai_config)
-        else:
-            client = OpenAI(**openai_config)
-        return client
+            self._clients.append(OpenAIClient(AzureOpenAI(**openai_config)))
+        elif api_type is None:
+            self._clients.append(OpenAIClient(OpenAI(**openai_config)))
+        # else a config for a custom client is set
+        # skipping until the register_custom_client is called with the appropriate class
+
+    def register_custom_client(self, ClientClass: Client, **kwargs):
```

Comment:
```suggestion
    def register_client(self, ClientClass: Client, **kwargs):
```

Reply:
all clients are created equal :D -- they are all put on a list and get picked anyway.

Reply:
lol fair point